### PR TITLE
Layer configurations to 3D views

### PR DESCRIPTION
**Pull request by jheijari**

### CODE REVIEW

File: server-extension/src/main/java/flyway/olcesium/V1_1__set_layers.java [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+        if (!layers.isEmpty()) {
+            updateCesiumViews(connection, layers);
+        }
+    }
+
+    private void updateNlsBaseLayer(Connection connection, OskariLayer layer) {
+        try {
+            // update
+            OskariLayerCapabilities caps = capabilitiesService.getCapabilities(layer);
+            if (caps != null) {
```
Comment (jampukka) [inline server-extension/src/main/java/flyway/olcesium/V1_1__set_layers.java]: `if (caps == null) { LOG.info(...); return; }` could make this function more readable by removing the nested `if`s

File: server-extension/src/main/java/flyway/olcesium/V1_1__set_layers.java [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+                }
+            }
+        } catch (Exception e) {
+            LOG.error(e, "Error setting selected layer for cesium views");
+        }
+    }
+
+    private List<View> getCesiumViews(Connection conn) throws SQLException {
+        List<View> list = new ArrayList<>();
+        final String sql = "SELECT id FROM portti_view WHERE name = '"+CESIUM_VIEW_NAME+"'";
```
Comment (jampukka) [inline server-extension/src/main/java/flyway/olcesium/V1_1__set_layers.java]: While there's no risk of SQL injection here, I'd still personally use `"... WHERE name = ?";`

File: server-extension/src/main/java/flyway/olcesium/V1_1__set_layers.java [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+                } else {
+                    LOG.warn("Updating Cesium views / mapfull bundle not found - view: ", view.getId());
+                }
+            }
+        } catch (Exception e) {
+            LOG.error(e, "Error setting selected layer for cesium views");
+        }
+    }
+
+    private List<View> getCesiumViews(Connection conn) throws SQLException {
```
Comment (jampukka) [inline server-extension/src/main/java/flyway/olcesium/V1_1__set_layers.java]: This method could also be `private List<Long> getCesiumViewIds(Connection conn) throws SQLException;`

File: server-extension/src/main/java/flyway/olcesium/V1_1__set_layers.java [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+    }
+
+    private void updateCesiumViews(Connection connection, List<OskariLayer> layers) {
+        try {
+            List<View> views = getCesiumViews(connection);
+            LOG.info("Updating Cesium views - count:", views.size());
+
+            JSONArray selectedLayers = new JSONArray();
+            JSONObject jsonLyr = new JSONObject();
+            jsonLyr.put("id", layers.get(0).getId());
```
[the remark this answers is not in the document]
Comment (jheijari) [reply server-extension/src/main/java/flyway/olcesium/V1_1__set_layers.java]: Layers (should be only two layers in this case 'taustakartta' and 'maastokartta') are base maps so instead adding them all to selectedlayers we add only one. Doesn't really matter which one.

File: server-extension/src/main/java/flyway/olcesium/V1_1__set_layers.java [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+    private void updateNlsBaseLayer(Connection connection, OskariLayer layer) {
+        try {
+            // update capabilities in cache
+            String data = CapabilitiesCacheService.getFromService(layer);
+            OskariLayerCapabilities caps = new OskariLayerCapabilities(
+                    layer.getSimplifiedUrl(true),
+                    layer.getType(),
+                    layer.getVersion(),
+                    data);
+            capabilitiesService.save(caps);
```
Comment (jampukka) [inline server-extension/src/main/java/flyway/olcesium/V1_1__set_layers.java]: This should be moved down a bit. We don't want to save the XML to the DB if WMTSCapabilitiesParser fails to parse it.

**Comment by jampukka** [comment]
CapabilitiesCacheService.save() needs to be called after we know the GetCapabilities response was parseable by our code. Other than that looks good.

**Pull request merged**